tests/test_pyvista.py

```diff
@@ -281,3 +281,23 @@ def test_imcache(cleanup_tester, verify_image_cache):
 
     result = testdir.runpytest("--fail_extra_image_cache")
     result.stdout.fnmatch_lines("*[Pp]assed*")
+
+
+def test_reset_only_failed(testdir):
+    """Test usage of the `reset_only_failed` flag."""
+    make_cached_images(testdir.tmpdir)
+    testdir.makepyfile(
+        """
+        import pyvista as pv
+        pv.OFF_SCREEN = True
+        def test_imcache(verify_image_cache):
+            sphere = pv.Box()
+            plotter = pv.Plotter()
+            plotter.add_mesh(sphere, color="blue")
+            plotter.show()
+        """
+    )
+
+    result = testdir.runpytest("--reset_only_failed")
+    result.stdout.fnmatch_lines("*[Pp]assed*")
```

Comment:
We should also test that the file was indeed overwritten, see https://github.com/pyvista/pytest-pyvista/blob/42f771b84fd99d98a1e202f432ee56a9773e8f05/tests/test_pyvista.py#L251

Reply:
Agree, resolved.